site/docs/references/mixin-syntax.md

```diff
@@ -1,124 +1,253 @@
+    -st-default: colorAndBg;
+}
+
+.codeMixedIn {
+    -st-mixin: colorAndBg(green, orange); 
+    font-family: monospace;
 }
-.a:hover {
-    background: white;
+```
+```css
+/* CSS output */
+.codeMixedIn {
+    color: green; /* from JS mixin */
+    background: orange; /* from JS mixin */
+    font-family: monospace; /* from local class */
 }
 ```
 
+### JavaScript mixins returning multiple rulesets
+
+Mixins can return multiple rulesets that are mixed into the target stylesheet. These rulesets can be written with the following syntax options:
+* `selector` - resulting ruleset is appended as a descendent selector to its mixed in target (in below example `.otherClass`) 
+* `&selector` - resulting ruleset references the parent selector into which it was mixed in (in below example `&:hover`, the parent selector is `.codeMixedIn`)
+
+
+```js
+/* file my-mixin.js */
+module.exports = function complexMixin([color, bgColor]){ 
+    /* arguments: array of string types */
+
+    return {
+        color: color,
+        background: bgColor,
+        "&:hover": {
+            color: "gold"
+        },
+        ".otherClass": {
+            color: "purple"
+        }
+    }
+};
+```
+
+``` css
+/* file example.st.css */
+:import {
+    -st-from: './my-mixin';
+    -st-default: complexMixin;
+}
+
+.codeMixedIn {
+    -st-mixin: complexMixin(green, orange); 
+    font-family: monospace;
+}
+```
 ```css
-/* CSS output*/
-.a {
-    color: red;
-    color: gold; /* added by golden */
-    background: black; /* added by golden */
-    background: white;
+/* CSS output */
+.codeMixedIn {
+    color: green; /* from JS mixin */
+    background: orange; /* from JS mixin */
+    font-family: monospace; /* from local class */
 }
-.a:hover {
-    color: black; /* added by golden */
-    background: gold; /* added by golden */
+
+.codeMixedIn:hover { /* from JS mixin with & */
+    color: gold; 
 }
-.a:hover {
-    background: white;
+
+.codeMixedIn .otherClass { /* from JS mixin with appended selector */
+    color: purple; 
+}
+```
+
+## How mixins are applied
+
+Mixins can add CSS declarations to the CSS ruleset to which they are applied.
+
+Rules are added at the position in the CSS where the `-st-mixin` is declared.
+Any selectors that are appended as a result of the mixin are added directly after the ruleset that the mixin was applied to.
+
+You can apply multiple mixins in either CSS or JavaScript, or both seperated by comma `-st-mixin: mixinA, mixinB`.
+Multiple mixins are applied according to the order that they are declared left to right.
+
+## Considerations when using mixins
+
+Take a look at these considerations before working with **Stylable** mixins.
+
+### Mixin usage with special characters
+You can escape special characters by wrapping them with quotes or using a backslash (`\`). 
+
+Example:
+```css
+.x { /* use quotations to include comma */
+    -st-mixin: mix(300, "xx,x"); /* ["300", "xx,x"] */
+}
+.y { /* escape slashes */
```

Comment:
this are not escape slashes. we use slash to escape special characters. here the escape is used to have quotation chars as part of the parameter